keyboards/strech/soulstone/keymaps/default/keymap.c

```diff
@@ -0,0 +1,56 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_ESC,   KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,             KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_DEL,   KC_BSPC,
+        KC_TAB,   KC_A,     KC_S,     KC_D,     KC_F,     KC_G,             KC_H,     KC_J,     KC_K,     KC_L,     KC_QUOT,  KC_ENT,
+        KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,             MO(3),    KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,  KC_RSFT,
+        KC_LCTL,  KC_LALT,                      MO(2),    MO(1),            KC_SPC,   KC_RALT,                                KC_RGUI,  KC_RCTL
+    ),
+    [1] = LAYOUT(
+        KC_GRV,   KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_UP,    KC_TRNS,  KC_MUTE,  KC_VOLD,  KC_VOLU,
+        KC_CAPS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_LEFT,  KC_DOWN,  KC_RGHT,  KC_SCLN,  KC_BSLS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,                      KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,                                KC_TRNS,  KC_TRNS
+    ),
+    [2] = LAYOUT(
+        KC_TRNS,  KC_1,     KC_2,     KC_3,     KC_4,     KC_5,             KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_MINS,  KC_EQL,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_P4,    KC_P5,    KC_P6,    KC_LBRC,  KC_RBRC,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_P1,    KC_P2,    KC_P3,    KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,                      KC_TRNS,  KC_TRNS,          KC_PENT,  KC_PDOT,                                KC_TRNS,  KC_TRNS
+    ),
+    [3] = LAYOUT(
+        KC_TRNS,  KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,            KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,                      KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,                                KC_TRNS,  KC_TRNS
+    )
+};
+
+bool led_update_user(led_t led_state) {
+    writePin(B1, led_state.caps_lock);
+    writePin(B2, led_state.num_lock);
+    return false;
+}
+
```

Comment:
```suggestion
```

Reply:
If I do that, led on B3 will not stay on high when I press MO keys. I've tried it and tested it. That's the reason I've returned it.

It's the same code as here https://github.com/qmk/qmk_firmware/blob/master/keyboards/primekb/prime_e/keymaps/default/keymap.c#L60-L72

Reply:
I've made a short video on how it's manifesting. At the beginning - I'm using current PR firmware, and then flash it with suggested changes.

As you can see - instead of lightning it just briefly goes on and off.

https://github.com/qmk/qmk_firmware/assets/920053/be023035-6e0d-49d0-bcd7-cd41916e2f5b

Reply:
I hope that helps @fauxpark

Reply:
If this LED is not meant as a scroll lock indicator, then you should not set it as such in the config.h. That's why you're getting this behaviour.

Reply:
Thanks @fauxpark I have the same understanding, but when I remove the code as you suggested and also remove B3 as scroll lock LED, for some reason, this LED gets less power, here is a video

```diff
diff --git a/keyboards/strech/soulstone/info.json b/keyboards/strech/soulstone/info.json
index 0710dff957..53037d1460 100644
--- a/keyboards/strech/soulstone/info.json
+++ b/keyboards/strech/soulstone/info.json
@@ -23,9 +23,8 @@
         "rows": ["D0", "D4", "B5", "F5"]
     },
     "indicators": {
-        "num_lock": "B1",
-        "caps_lock": "B2",
-        "scroll_lock": "B3"
+        "caps_lock": "B1",
+        "num_lock": "B2"
     },
     "dynamic_keymap": {
         "layer_count": 6
```

and

```diff
diff --git a/keyboards/strech/soulstone/keymaps/default/keymap.c b/keyboards/strech/soulstone/keymaps/default/keymap.c
index 7e3ea67096..3f9691b7af 100644
--- a/keyboards/strech/soulstone/keymaps/default/keymap.c
+++ b/keyboards/strech/soulstone/keymaps/default/keymap.c
@@ -43,11 +43,11 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
     )
 };
 
-bool led_update_user(led_t led_state) {
-    writePin(B1, led_state.caps_lock);
-    writePin(B2, led_state.num_lock);
-    return false;
-}
+/* bool led_update_user(led_t led_state) { */
+/*     writePin(B1, led_state.caps_lock); */
+/*     writePin(B2, led_state.num_lock); */
+/*     return false; */
+/* } */
 
 // Function for layer indicator LED
 layer_state_t layer_state_set_user(layer_state_t state) {
```

https://github.com/qmk/qmk_firmware/assets/920053/ffd61141-f166-41d6-998d-1db3603b03fb

Reply:
I find that if I add `setPinOutput` to that function it shines with the same brightness

```diff
diff --git a/keyboards/strech/soulstone/keymaps/default/keymap.c b/keyboards/strech/soulstone/keymaps/default/keymap.c
index 7e3ea67096..4f2be37945 100644
--- a/keyboards/strech/soulstone/keymaps/default/keymap.c
+++ b/keyboards/strech/soulstone/keymaps/default/keymap.c
@@ -43,14 +43,15 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
     )
 };
 
-bool led_update_user(led_t led_state) {
-    writePin(B1, led_state.caps_lock);
-    writePin(B2, led_state.num_lock);
-    return false;
-}
- 
 // Function for layer indicator LED
 layer_state_t layer_state_set_user(layer_state_t state) {
+    setPinOutput(B3);
     writePin(B3, !layer_state_cmp(state, 0));
     return state;
 }
```

is it a correct way? @fauxpark